ext/standard/var_unserializer.c

```diff
@@ -349,9 +354,40 @@ static zend_always_inline int process_nested_data(UNSERIALIZE_PARAMETER, HashTab
 				if ((old_data = zend_hash_find(ht, Z_STR(key))) != NULL) {
 					if (Z_TYPE_P(old_data) == IS_INDIRECT) {
 						old_data = Z_INDIRECT_P(old_data);
+						if (Z_STRVAL(key)[0] == 0) {
+							zend_string *member, *class;
+							const char *class_name, *prop_name;
+							size_t prop_name_len;
+							zend_unmangle_property_name_ex(Z_STR(key), &class_name, &prop_name, &prop_name_len);
+							member = zend_string_init(prop_name, prop_name_len, 0);
+							class = zend_string_init(class_name, strlen(class_name), 0);
+							zend_str_tolower(ZSTR_VAL(class), ZSTR_LEN(class));
+							EG(fake_scope) = class_name[0] == '*' ? ce : zend_hash_find_ptr(EG(class_table), class);
+							info = zend_get_property_info(EG(fake_scope), member, 1);
+							EG(fake_scope) = NULL;
+							zend_string_release(member);
+							zend_string_release(class);
+						} else {
+							info = zend_get_property_info(ce, Z_STR(key), 1);
+						}
+						var_push_dtor(var_hash, old_data);
```

Comment:
Someone familiar with serialize vulnerabilities should take a look at this. Maybe Stas?